asmcomp/cmmgen.ml

```diff
@@ -64,6 +64,13 @@ let add_unboxed_id id unboxed_id bn env =
     unboxed_ids = V.add id (unboxed_id, bn) env.unboxed_ids;
   }
 
+let constants_tbl = Hashtbl.create 16
```

Comment:
I agree on principle, but I'll go with the suggestion to use Cmmgen_state.  It would make sense to move all of it into an environment passed through Cmmgen (there is already an Cmmgen.env, but the extra plumbing is still a bit invasive), but this can be done later.